numpy/_core/tests/test_hashtable.py

```diff
@@ -23,8 +23,3 @@ def test_identity_hashtable(key_length, length):
 
     res = identityhash_tester(key_length, keys_vals, replace=True)
     assert res is expected
-
-    # check that ensuring one duplicate definitely raises:
-    keys_vals.insert(0, keys_vals[-2])
-    with pytest.raises(RuntimeError):
-        identityhash_tester(key_length, keys_vals)
```

Comment:
Given the change in the semantics in this PR, I don't think this bit makes sense anymore. I don't think this property is used anywhere except for this test.

Reply:
I'm a bit confused: Why not just change it with something that definitely is different and something that is definitely the same?

Reply:
I added an `pytest.raises(RuntimeError)` test where I pass it identical keys with distinct values, see latest commit. Let me know if you had something additional in mind, I had a little trouble parsing this comment.

Reply:
I am pretty confident that is the point :), going to merge, thanks.

Reply:
This was indeed the point! Thanks, all